happi/cli.py

```diff
@@ -102,13 +105,58 @@ def happi_cli(args):
                 return
             if value.replace('.', '').isnumeric():
                 logger.debug('Changed %s to float', value)
-                value = float(value)
-            client_args[criteria] = value
+                value = str(float(value))
 
-        results = client.search(**client_args)
-        if results:
+            if criteria == 'z' and ',' in value:
```

Comment:
I wonder if there is a way to generalize this more, since in the future we may have other numeric fields that want ranges. Perhaps other position fields? Perhaps other use cases we can't really imagine?

Generalizing this doesn't have to be in-scope for this PR.

Reply:
Yeah i agree with you here, i should have a more generalized way to look for ranges. I thought about having only a comma `,` but I don't think it would be sufficient, just in case we have commas in some of our fields.

I need to think about this one....

Reply:
I agree, having a comma is not enough. It's unfortunate that the cli at this point doesn't have a clear API for checking what kind of field a specific criteria is supposed to be, since we may be matching fields across different container classes... If it could just look up what type `criteria` was supposed to be based on the `EntryInfo`, that would be a clean way to decide if we can try to interpret the field as a range.

Something like splitting on the comma and checking for two numeric quantities seems really distasteful but could work...

Reply:
Yeah i like your idea of being able to look up what type `criteria` was supposed to be based on the `EntryInfo` somehow... for now i basically implemented your suggestion of splitting on the comma and checking - but with a bit of a tweak, please let me know if this is ok: 
instead of checking for two numeric quantities i also split on dot `.` as well (if they are floats) and make a big number out of these two and check if this is numeric..... am I covering all the scenarios here, what do you think? :)

Reply:
I changed this a bit in my last commit 8caebba - please let me know what you think